implementations/python-jsonschema/bowtie_jsonschema.py

```diff
@@ -3,14 +3,46 @@
 
 from dataclasses import dataclass
 from importlib import metadata
+from itertools import zip_longest
 from typing import TYPE_CHECKING
 import json
 import platform
 import sys
 import traceback
 
 from jsonschema.validators import validator_for
-import referencing.jsonschema
+
+jsonschema_version = metadata.version("jsonschema")
+
+
+def compare_versions(v1: str, v2: str) -> int:
```

Comment:
Here jsonschema *does* use Python versioning (and semver) so we have a single known implementation and version scheme -- so here we definitely should use an existing parser (probably the one you had from the packaging repo).